File: output/ChatGPT/codenet/C++/Python/s391929974.py
while True:
    n = int(input())
    if n == 0:
        break

    sq = [0] * 10000
    fr = [0] * 5000
    com = []

    for i in range(1, n):
        sq[i] = (i*i) % n

    for i in range(1, n):
        f = True
        for j in range(len(com)):
            if com[j] == sq[i]:
                f = False
                break
        if f:
            com.append(sq[i])

    for i in range((n-1)//2+1):
        fr[i] = 0

    for i in range(len(com)):
        for j in range(len(com)):
            if i != j:
                tmp = com[i] - com[j]
                if tmp < 0:
                    tmp += n
                if tmp > (n-1)//2:
                    tmp = n - tmp
                fr[tmp] += 1

    for i in range(1, (n-1)//2+1):
        print(fr[i])

File: output/ChatGPT/codenet/C++/Java/s391929974.java
import java.util.ArrayList;
import java.util.Scanner;

public class s391929974 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (true) {
            int n = scanner.nextInt();
            if (n == 0) break;

            int[] sq = new int[10000];
            int[] fr = new int[5000];
            ArrayList<Integer> com = new ArrayList<>();
            int tmp;

            for (int i = 1; i < n; i++) {
                sq[i] = (i * i) % n;
            }

            com.clear();

            for (int i = 1; i < n; i++) {
                boolean f = true;
                for (int j = 0; j < com.size(); j++) {
                    if (com.get(j) == sq[i]) {
                        f = false;
                        break;
                    }
                }
                if (f) com.add(sq[i]);
            }

            for (int i = 0; i <= (n - 1) / 2; i++) {
                fr[i] = 0;
            }

            for (int i = 0; i < com.size(); i++) {
                for (int j = 0; j < com.size(); j++) {
                    if (i != j) {
                        tmp = com.get(i) - com.get(j);
                        if (tmp < 0) tmp += n;
                        if (tmp > (n - 1) / 2) tmp = n - tmp;
                        fr[tmp]++;
                    }
                }
            }

            for (int i = 1; i <= (n - 1) / 2; i++) {
                System.out.println(fr[i]);
            }
        }
    }
}